### ISSUE TITLE
Adding Fig's SSH Integration creates issues with `git` authentication

### ISSUE THREAD

**Issue by zacKolton** [withfig/fig#326]
### Description:
> I have the SSH integration added, why can't I push to my companies self-hosting gitlab instance?

Okay so first off, great job, this app is so sweet. Honestly, the best terminal app out there. 

Nevertheless, I ran into this when setting up fig. Essentially when going through the setup (before cli tutorial), it asks if you it can edit your `.ssh/config` file or skip for later. Me unknowingly and way too excited, of course I allowed fig to. 

Right after I enabled this I couldn't pull/push to/from my local repos because the remotes couldn't be found / interacted with. So anyways, I uninstalled fig and reinstalled it without editing the ssh folder and it now works just like before!

So yeah, im not very adverse in SSH connections, but it was something to do with that messing up my connection with my company's gitlab instance, I presume authentication.  I thought I'd bring this up in-case anyone else had the same scenario. 
 
### Details:
|macOS|Fig|Shell|
|-|-|-|
|11.3.0|Version 1.0.46 (B250)|/bin/zsh|
<details><summary><code>fig diagnostic</code></summary>
<p>
<pre>Version 1.0.46 (B250) [Canadian English] 
UserShell: /bin/zsh
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/zackolton/.fig/bin/fig
Accessibility: true
Number of specs: 155
SSH Integration: false
Tmux Integration: true
Keybindings path: /Users/zackolton/.fig/user/keybindings
iTerm Integration: false 
Hyper Integration: false
VSCode Integration: true
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Users/zackolton/.fig/bin
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: /bin/zsh (8173) - ttys012
Current working directory: /Users/zackolton/repos/report-templates
Current window identifier: 7809/% (com.apple.Terminal)</pre>
</p>
</details>

**Comment by jeremybarbet** [comment]
I second that, I pressed all the buttons during the setup screen and it ended up changing the `.ssh/config` file, which resulted into a `fatal: protocol error: bad line length character: Usin` when trying to do a `git push`. 

Workaround was to diasbled the ssh integration and revert the changes done to the config file.

```diff
...

- # Fig SSH Integration: Enabled
- Include ~/.fig/ssh
- # End of Fig SSH Integration
```

|macOS|Fig|Shell|
|-|-|-|
|11.4.0|Version 1.0.46 (B250)|/usr/local/bin/fish|
<details><summary><code>fig diagnostic</code></summary>
<p>
<pre>Version 1.0.46 (B250) [French] 
UserShell: /usr/local/bin/fish
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/jeremy/.fig/bin/fig
Accessibility: true
Number of specs: 158
SSH Integration: true
Tmux Integration: true
Keybindings path: /Users/jeremy/.fig/user/keybindings
iTerm Integration: true [Authenticated]
Hyper Integration: false
VSCode Integration: true
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: Using
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: /usr/local/Cellar/fish/3.1.2/bin/fish (12163) - ttys003
Current working directory: /Users/jeremy/www/[REDACTED]
Current window identifier: 426/[REDACTED]% (com.googlecode.iterm2)</pre>
</p>
</details>

**Comment by mschrage** [comment]
@jeremybarbet @zacKolton thank you both for reporting this. I'm not totally sure why this is happening. 

Is there anything we can do to make it more clear what's going on during the onboarding?

**Comment by zacKolton** [comment]
Honestly i just clicked it because I'm like "eh probably". I didn't have an example of why *not* to use it.

Maybe you could add an example of why you would want it and why you wouldn't.

Why did you add that feature in the first place? Would be an okay start lol

**Comment by tlaak** [comment]
I just bumped into this same issue. Fig version 1.0.55 (B365) on macOS Monterey 12.2.1, fish shell version 3.3.1.

Any remote git operation (e.g. pull, push) will fail with the error: `fatal: protocol error: bad line length character: ?]69` and it took me a good while to be able to figure out the cause!

After removing the lines added by Fig from my `.ssh/config`, the problem got resolved.

Running `ssh git@github.com git-receive-pack path-to/repository.git` did not reveal anything useful. I'm not sure what that character `?]69` is, but in other similar errors, the characters showing up are usually the first characters of the error message coming from ssh.
